src/main/java/software/amazon/event/kafkaconnector/mapping/DefaultEventBridgeMapper.java

```diff
@@ -0,0 +1,133 @@
+    try {
+      return success(
+          record,
+          PutEventsRequestEntry.builder()
+              .eventBusName(config.eventBusArn)
+              .source(sourcePrefix + config.connectorId)
+              .detailType(config.getDetailType(record.topic()))
+              .resources(config.resources)
+              .detail(createJsonPayload(record))
+              .build());
+    } catch (JsonProcessingException e) {
+      return failure(
+          record,
+          EventBridgeResult.Error.of(
+              "Cannot convert Kafka record to EventBridge event: " + e.getMessage(), 1));
+    }
+  }
+
+  private String createJsonPayload(SinkRecord record) throws JsonProcessingException {
+    var root = objectMapper.createObjectNode();
+    root.put("topic", record.topic());
+    root.put("partition", record.kafkaPartition());
+    root.put("offset", record.kafkaOffset());
+    root.put("timestamp", record.timestamp());
+    root.put("timestampType", record.timestampType().toString());
+    root.putIfAbsent("headers", createHeaderArray(record));
+
+    if (record.key() == null) {
+      root.putIfAbsent("key", null);
+    } else {
+      root.putIfAbsent(
+          "key",
+          createJSONFromByteArray(
+              jsonConverter.fromConnectData(record.topic(), record.keySchema(), record.key())));
+    }
+
+    // tombstone handling
+    if (record.value() == null) {
+      root.putIfAbsent("value", null);
+    } else {
+      root.putIfAbsent(
+          "value",
+          createJSONFromByteArray(
+              jsonConverter.fromConnectData(record.topic(), record.valueSchema(), record.value())));
+    }
+
+    return root.toString();
+  }
+
+  /**
+   * This method serializes Kafka message headers to JSON.
+   *
+   * @param record Kafka record to be sent to EventBridge
+   * @return headers to be added to EventBridge message
+   * @throws JsonProcessingException
+   */
+  private ArrayNode createHeaderArray(SinkRecord record) throws JsonProcessingException {
+    var headersArray = objectMapper.createArrayNode();
+
+    for (Header header : record.headers()) {
+      var headerItem = objectMapper.createObjectNode();
+      headerItem.putIfAbsent(
+          header.key(),
+          createJSONFromByteArray(
+              jsonConverter.fromConnectHeader(
+                  record.topic(), header.key(), header.schema(), header.value())));
+      headersArray.add(headerItem);
+    }
+    return headersArray;
+  }
+
+  /**
+   * This method converts the byteArray which is returned by the {@link JsonConverter} to JSON.
+   *
+   * @param jsonBytes - byteArray to convert to JSON
+   * @return the JSON representation of jsonBytes
+   * @throws JsonProcessingException
+   */
+  private JsonNode createJSONFromByteArray(byte[] jsonBytes) throws JsonProcessingException {
+    return objectMapper.readTree(new String(jsonBytes, StandardCharsets.UTF_8));
```

Comment:
Question: we could directly read from `byte[]` https://fasterxml.github.io/jackson-databind/javadoc/2.7/com/fasterxml/jackson/databind/ObjectMapper.html#readTree(byte[])?

Reply:
Have to check.

Reply:
Did a quick smoke test: https://github.com/embano1/eventbridge-kafka-connector/pull/4/files
Guess it's safe, would have to verify with Avro, but should also work due to the `jsonConverter` used internally